src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -70,7 +83,49 @@ class RedownloadService[F[_]](
       }
     } yield proposals
 
-  def fetchMissingSnapshots: F[SnapshotsAtHeight] = ??? // TODO: Implement
+  def fetchStoredSnapshots: F[Map[APIClient, Seq[String]]] =
+    for {
+      peers <- cluster.getPeerInfo.map(_.values.toList)
+      apiClients = peers.map(_.client)
+      responses <- apiClients.traverse { client =>
+        client
+          .getNonBlockingF[F, Seq[String]]("/snapshot/stored")(C)
```

Comment:
I think you should remove the first `/` as we're not using it like that in other places